llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/retriever.py

```diff
@@ -58,7 +58,9 @@ def _result_nodes_to_node_with_score(
     def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
         """Retrieve from the platform."""
         pipelines = self._client.pipeline.search_pipelines(
-            project_name=self.project_name, pipeline_name=self.name
+            project_name=self.project_name,
+            pipeline_name=self.name,
+            pipeline_type=PipelineType.MANAGED,
         )
         if len(pipelines) != 1:
```

Comment:
should we make this if-else check more strict too? if len(pipelines) == 0 vs. len(pipelines) > 1

Reply:
technically the `len(pipelines) > 1` case should never happen now with the `pipeline_type` filter, but can add that `elif` just in case